### PR TITLE
Base test classes

### PR DESCRIPTION
**Pull request by jomora**
This PR is the first one of a sequence of PRs to cope with the complexity introduced in #474. 

This PR introduces `class BaseTest`, into the test class hierarchy. This introduction requires to change the base class of `ComponentTest, ModuleTest, SystemTest` and `SubsystemTest` to `BaseTest`.

The last commit in this PR solves compilation errors of concrete test implementations and adapts them to adhere to the assumptions stated in the JavaDoc of `BaseTest`.

To verify that tests are still work properly, `mvn clean install` has successfully been run.

### CODE REVIEW

File: modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ *     super.doSetUp();
+ *     // ... my code
+ *   }
+ * }
+ * </pre>
+ *
+ * @author shuber, jmolinar
+ */
+public abstract class BaseTest extends Assertions {
+
```
Comment (hohwille) [inline modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java]: here we need the static variable to determine if we are triggered for the first time in this class to distinguish between setUp() first type and subsequent calls. You should have all the code from me how to solve this problem. Just paste it into this class.

File: modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  @After
+  public final void tearDown() {
+
+    doTearDown();
+  }
+
+  /**
+   * Provides initialization previous to the creation of the text fixture.
+   */
+  protected void doSetUp() {
```
Comment (hohwille) [inline modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java]: Either we add a boolean parameter `initialSetup`  or we provide a boolean isInitialSetup() method.

**Comment by jomora** [comment]
@hohwille 
> Or is AbstractRestSerivceTest already in develop because someone accidently merged? I did not check as I am hopeless out of time.

`AbstractRestServiceTest` has been added to `develop` a [few months ago](https://github.com/oasp/oasp4j/commit/00d433c6015ff104a58552ef36ca937eb7697a56).
However, I did not adapt it for the current PR since no change is necessary, because `doSetup` and `doTeardown` are overriden as expected.
Do you agree?

Of course, in subsequent PRs changes will occur as proposed in #474.

File: modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java (hunk 1 above)
Comment (jomora) [inline modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java]: I added the static variable, and an according test class. Please refer to the newly added commits.

File: modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java (hunk 2 above)
Comment (jomora) [inline modules/test/src/main/java/io/oasp/module/test/common/base/BaseTest.java]: I added a boolean parameter and test it's working in `BaseTestTest` class. Please review my changes

**Comment by hohwille** [comment]
All agreed. However, could someone else also give us feedback. After rethinking IMHO adding a method instead of the boolean parameter would be better. WDYT?

**Comment by kiran-vadla** [comment]
@jomora , from the history , I see that you have worked on it . Hence assigning it to you. 

Thanks,
Kiran.

**Comment by kiran-vadla** [comment]
@jomora , I will take up this PR

**Comment by kiran-vadla** [comment]
@jomora , as mentioned by you , this PR along with PRs #491, #474 and Issues #522 and #385 will be on hold until discussion with Devonfw community to receive inputs from other experienced archtiects is completed. 

Thanks,
@kiran-vadla

**Comment by kiran-vadla** [comment]
@jomora , FYI, this PR has to be merged onto develop-2.3.0.

Thanks,
@kiran-vadla

**Pull request merged**